File: python/program/Toko.py
class Toko:
    def __init__(self, id_toko="", nama_toko="", alamat=""):
        self.id_toko = id_toko
        self.nama_toko = nama_toko
        self.alamat = alamat
        self.daftar_transaksi = []

    # Getter
    def get_id_toko(self):
        return self.id_toko

    def get_nama_toko(self):
        return self.nama_toko

    def get_alamat(self):
        return self.alamat

    # Setter
    def set_id_toko(self, id_toko):
        self.id_toko = id_toko

    def set_nama_toko(self, nama_toko):
        self.nama_toko = nama_toko

    def set_alamat(self, alamat):
        self.alamat = alamat

    # Method
    def tambah_transaksi(self, transaksi):
        self.daftar_transaksi.append(transaksi)

    # Getter daftar transaksi
    def get_daftar_transaksi(self):
        return self.daftar_transaksi

File: cpp/program/Toko.cpp
#pragma once
#include <string>
#include <vector>
#include "Transaksi.cpp"
using namespace std;

class Toko {
private:
    string id_toko;
    string namaToko;
    string alamat;
    vector<Transaksi*> daftarTransaksi;
public:
    Toko(string id="", string nama="", string alamat="") {
        this->id_toko = id;
        this->namaToko = nama;
        this->alamat = alamat;
    }

    // Getter & Setter
    string getIdToko() { return id_toko; }
    string getNamaToko() { return namaToko; }
    string getAlamat() { return alamat; }

    void setIdToko(string id) { id_toko = id; }
    void setNamaToko(string n) { namaToko = n; }
    void setAlamat(string a) { alamat = a; }

    // Method
    void tambahTransaksi(Transaksi* t) {
        daftarTransaksi.push_back(t);
    }

    // Getter daftar transaksi
    vector<Transaksi*> getDaftarTransaksi() {
        return daftarTransaksi;
    }
};
